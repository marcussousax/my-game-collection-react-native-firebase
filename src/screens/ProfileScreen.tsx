import React from 'react'
import { Button, Text, TouchableOpacity, View } from 'react-native'
import { AuthContext } from '../contexts/auth'
import { getDocRef } from '../services/api'
import { AppStackParamList, GameProps } from '../types'
import { StackScreenProps } from '@react-navigation/stack'

const ProfileScreen = ({ navigation }: StackScreenProps<AppStackParamList>) => {
    const { user, signOut } = React.useContext(AuthContext)
    const [listGames, setListGames] = React.useState<GameProps[]>([])
    const [loading, setLoading] = React.useState<boolean>(true)

    React.useEffect(() => {
        getDocRef('games').onSnapshot(QuerySnapshot => {
            const gamesSnapshot:
                | ((prevState: never[]) => never[])
                | GameProps[] = []
            QuerySnapshot.docs.forEach(doc => {
                const { name, status, system } = doc.data()
                gamesSnapshot.push({
                    id: doc.id,
                    name,
                    status,
                    system
                })
            })

            setListGames(gamesSnapshot)
            setLoading(false)
        })
    }, [])

    return (
        <View>
            <Text>{user?.displayName}</Text>
            <Text>{user?.email}</Text>

            {loading ? (
                <Text>loading games</Text>
            ) : (
                listGames.map(game => (
                    <TouchableOpacity key={game.id} onPress={() => null}>
                        <Text>{game.name}</Text>
                        <Text>{game.system}</Text>
                        <Text>{game.status}</Text>
                    </TouchableOpacity>
                ))
            )}

            <Button
                title={'Add Game'}
                onPress={() => navigation.navigate('AddGameScreen')}
            />
            <Button title={'Logout'} onPress={() => signOut()} />
        </View>
    )
}

export default ProfileScreen
