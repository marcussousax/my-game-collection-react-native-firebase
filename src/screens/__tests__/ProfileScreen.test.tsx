import 'react-native'
import React from 'react'
import ProfileScreen from '../ProfileScreen'
import TestRenderer from 'react-test-renderer'

describe('<ProfileScreen>', () => {
    it('it renders correctly', () => {
        const comp = TestRenderer.create(<ProfileScreen />).toJSON()
        expect(comp).toMatchSnapshot()
    })
})
