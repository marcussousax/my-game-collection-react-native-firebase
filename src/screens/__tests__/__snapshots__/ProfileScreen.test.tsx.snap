// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileScreen> it renders correctly 1`] = `
Array [
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#02cbd3",
        "height": 100,
        "justifyContent": "flex-end",
        "left": 0,
        "marginBottom": 150,
        "right": 0,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "borderRadius": 60,
          "height": 90,
          "marginTop": 40,
          "position": "absolute",
          "top": 10,
          "width": 90,
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 30,
          "position": "absolute",
          "top": 150,
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 15,
          "position": "absolute",
          "top": 190,
        }
      }
    />
  </View>,
  <RCTScrollView>
    <View>
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Logout
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>,
]
`;
